src/model/CoilCoolingDX.cpp

```diff
@@ -0,0 +1,345 @@
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "CoilCoolingDX.hpp"
+#include "CoilCoolingDX_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+#include "Schedule.hpp"
+#include "Schedule_Impl.hpp"
+#include "ThermalZone.hpp"
+#include "ThermalZone_Impl.hpp"
+#include "ScheduleTypeLimits.hpp"
+#include "ScheduleTypeRegistry.hpp"
+#include "CoilCoolingDXCurveFitPerformance.hpp"
+#include "CoilCoolingDXCurveFitPerformance_Impl.hpp"
+
+#include <utilities/idd/IddFactory.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/OS_Coil_Cooling_DX_FieldEnums.hxx>
+#include "../utilities/units/Unit.hpp"
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  CoilCoolingDX_Impl::CoilCoolingDX_Impl(const IdfObject& idfObject,
+                                         Model_Impl* model,
+                                         bool keepHandle)
+    : StraightComponent_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == CoilCoolingDX::iddObjectType());
+  }
+
+  CoilCoolingDX_Impl::CoilCoolingDX_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                         Model_Impl* model,
+                                         bool keepHandle)
+    : StraightComponent_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == CoilCoolingDX::iddObjectType());
+  }
+
+  CoilCoolingDX_Impl::CoilCoolingDX_Impl(const CoilCoolingDX_Impl& other,
+                                         Model_Impl* model,
+                                         bool keepHandle)
+    : StraightComponent_Impl(other,model,keepHandle)
+  {}
+
+  const std::vector<std::string>& CoilCoolingDX_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
```

Comment:
fill it.

```
"Cooling Coil Total Cooling Rate",
"Cooling Coil Total Cooling Energy",
"Cooling Coil Sensible Cooling Rate",
"Cooling Coil Sensible Cooling Energy",
"Cooling Coil Latent Cooling Rate",
"Cooling Coil Latent Cooling Energy",
"Cooling Coil Electric Power",
"Cooling Coil Electric Energy",
"Cooling Coil Runtime Fraction",
"SubcoolReheat Cooling Coil Operation Mode",
"SubcoolReheat Cooling Coil Operation Mode Ratio",
"SubcoolReheat Cooling Coil Recovered Heat Energy Rate",
"SubcoolReheat Cooling Coil Recovered Heat Energy"
```

source: https://bigladdersoftware.com/epx/docs/9-3/input-output-reference/group-coil-cooling-dx.html#outputs-01